code/datums/diseases/advance/symptoms/narcolepsy.dm

```diff
@@ -18,76 +18,50 @@ Bonus
 	stealth = -1
 	resistance = -2
 	stage_speed = -3
-	transmittable = -4
+	transmittable = 0
 	level = 6
-	symptom_delay_min = 15
-	symptom_delay_max = 80
+	symptom_delay_min = 25
+	symptom_delay_max = 70
 	severity = 4
-	var/sleep_level = 0
-	var/sleepy_ticks = 0
-	var/stamina = FALSE
-	threshold_desc = "<b>Transmission 7:</b> Also relaxes the muscles, weakening and slowing the host.<br>\
+	var/yawning = FALSE
+	threshold_desc = "<b>Transmission 7:</b> Causes the host to occasionally emit a yawn that spreads the virus in a manner similar to that of a sneeze.<br>\
 					  <b>Resistance 10:</b> Causes narcolepsy more often, increasing the chance of the host falling asleep."
 
 /datum/symptom/narcolepsy/Start(datum/disease/advance/A)
 	if(!..())
 		return
-	if(A.properties["transmittable"] >= 7) //stamina damage
-		stamina = TRUE
+	if(A.properties["transmittable"] >= 7) //yawning (mostly just some copy+pasted code from sneezing, with a few tweaks)
+		yawning = TRUE
 	if(A.properties["resistance"] >= 10) //act more often
-		symptom_delay_min = 10
-		symptom_delay_max = 60
+		symptom_delay_min = 20
+		symptom_delay_max = 45
 
 /datum/symptom/narcolepsy/Activate(var/datum/disease/advance/A)
 	var/mob/living/M = A.affected_mob
-	//this ticks even when on cooldown
-	switch(sleep_level) //Works sorta like morphine
-		if(10 to 19)
-			M.drowsyness += 1
-		if(20 to INFINITY)
-			M.Sleeping(30, 0)
-			sleep_level = 0
-			sleepy_ticks = 0
-
-	if(sleepy_ticks && A.stage>=5)
-		sleep_level++
-		sleepy_ticks--
-	else
-		sleep_level = 0
-
-	if(!..())
-		return
-
 	switch(A.stage)
 		if(1)
-			if(prob(10))
+			if(prob(50))
 				to_chat(M, "<span class='warning'>You feel tired.</span>")
 		if(2)
-			if(prob(10))
+			if(prob(50))
 				to_chat(M, "<span class='warning'>You feel very tired.</span>")
-				sleepy_ticks += rand(10,14)
-				if(stamina)
-					M.adjustStaminaLoss(10)
 		if(3)
-			if(prob(15))
+			if(prob(50))
 				to_chat(M, "<span class='warning'>You try to focus on staying awake.</span>")
-				sleepy_ticks += rand(10,14)
-				if(stamina)
-					M.adjustStaminaLoss(15)
+			M.drowsyness += 5
 		if(4)
-			if(prob(20))
+			if(prob(50))
 				to_chat(M, "<span class='warning'>You nod off for a moment.</span>")
-				sleepy_ticks += rand(10,14)
-				if(stamina)
-					M.adjustStaminaLoss(20)
+			M.drowsyness += 10
+			if(yawning)
+				M.emote("yawn")
+				if(M.CanSpreadAirborneDisease())
+					A.spread(6)
 		if(5)
-			if(prob(25))
+			if(prob(50))
```

Comment:
If all of them have the same prob() just move the if(prob(50)) in front of the switch.

Reply:
But then the drowsyness-adding code and yawning code won't run, no?

Are you aksing me to set a variable to prob(50) on each run through of Activate() and use that?